lib/uplinkc/testdata/object_test.c

```diff
@@ -187,6 +187,36 @@ void handle_project(ProjectRef project) {
             free_downloader(downloader);
         }
 
+        { // download range
+            long start = 100;
+            long limit = 1024 * 4;
+            DownloaderRef downloader = download_range(bucket, object_paths[i], start, limit, err);
+            require_noerror(*err);
+
+            uint8_t *downloaded_data = malloc(data_len);
+            memset(downloaded_data, '\0', data_len);
+            size_t downloaded_total = 0;
+
+            size_t size_to_read = 256 + i;
+            while (downloaded_total < data_len) {
+                size_t read_size = download_read(downloader, &downloaded_data[downloaded_total], size_to_read, err);
+                require_noerror(*err);
+
+                if (read_size == 0) {
+                    break;
+                }
+
+                downloaded_total += read_size;
+            }
+
+            download_close(downloader, err);
+            require_noerror(*err);
+            require(memcmp(&data[start], downloaded_data, limit) == 0);
```

Comment:
Let's check that `downloaded_total` is equal to `limit`

Reply:
done